### PR TITLE
fix(highlights): Fixes #1553 Limit highlights to one item per domain

### LINKED ISSUE #1553: Limit highlights to one item per domain (weighted highlights)
I get duplicate highlights often on the new highlights view. 

Using version Activity Stream Dev 1.1.4-dev-1476371709

screen shot 2016-10-13 at 2 03 10 pm

### PR DESCRIPTION
**Pull request by sarracini**
- Dedupes the sites based on a key (in this case, domain)
- Update tests

### CODE REVIEW

File: common/recommender/Baseline.js [hunk 1]
```diff
@@ -317,6 +318,24 @@ class Baseline {
...
     return entries;
   }
+
+  _dedupeSites(entries) {
+    let dedupedEntries = new Map();
+    entries.forEach(item => {
+      let key = this._createDedupeKey(item);
+      if (!dedupedEntries.get(key)) {
+        dedupedEntries.set(key, [item]);
+      }
```
Comment (jaredkerim) [inline common/recommender/Baseline.js]: You forgot the case where the key exists and you want to append.

File: common/recommender/Baseline.js [hunk 2]
```diff
@@ -317,6 +318,24 @@ class Baseline {
...
+    });
+
+    let results = [];
+    dedupedEntries.forEach(item => results.push(item[0]));
```
Comment (jaredkerim) [inline common/recommender/Baseline.js]: You need to sort these by weight and take the first.

Comment (sarracini) [inline common/recommender/Baseline.js]: they come in sorted order, so the first one we encounter will already have the greatest weight

**Comment by k88hudson** [comment]
LGTM

File: common/recommender/Baseline.js [hunk 3]
```diff
@@ -317,6 +318,24 @@ class Baseline {
 
     return entries;
   }
+
+  _dedupeSites(entries) {
```
Comment (k88hudson) [inline common/recommender/Baseline.js]: you might want to rename to `dedupeSitesByHost` to be clearer

Comment (sarracini) [inline common/recommender/Baseline.js]: I was thinking leaving it blank so that if we ever change the dedupe key we can still use the generic method

**Pull request merged**